src/models/transfers.js

```diff
@@ -231,24 +233,26 @@ function * fulfillTransfer (transferId, fulfillment) {
 
     // We don't know if notary is executing or rejecting transfer
     const isValidExecution = Boolean(transfer.execution_condition &&
-      Condition.testFulfillment(transfer.execution_condition, fulfillment))
+      transfer.execution_condition.type === fulfillment.getData().condition_fulfillment.type &&
+      Condition.testFulfillment(transfer.execution_condition, fulfillment.getData().condition_fulfillment))
```

Comment:
I'm not opposed to re-working the logic in the condition library and making corresponding changes to this code, but until we do that, we need to check that the fulfillment type matches the condition type, or `five-bells-condition` may throw an error. I also think re-working the checking of conditions is a task for another (hopefully much smaller) PR. We can open a ticket for that, if you'd like.